src/node_watchdog.cc

```diff
@@ -100,16 +124,55 @@ SigintWatchdog::SigintWatchdog(
   SigintWatchdogHelper::GetInstance()->Start();
 }
 
+SigintWatchdog::SigintWatchdog(
+  v8::Isolate* isolate, Local<v8::Function> interrupt)
+    : isolate_(isolate) {
+  ref.Reset(isolate, interrupt);
+  auto env = node::Environment::GetCurrent(isolate);
+  loop_activate = new uv_async_t;
+  loop_activate->data = this;
+  // Just activate the loop, doesn't do callback.
+  uv_async_init(env->event_loop(), loop_activate, [](uv_async_t* it){
+    SigintWatchdog* self = reinterpret_cast<SigintWatchdog*>(it->data);
+    self->HandleInterrupt();
+  });
```

Comment:
Can you `CHECK_EQ(0, uv_async_init(...))` the return value? It's unlikely but not impossible for it to fail.